### ISSUE TITLE
How to ensure universal sound source localization？

### ISSUE THREAD

**Issue by lchen1019** [ruohaoguo/ovavss#5]
Hi,

I am very interested in your work. I noticed that OV-AVSS first relies on a universal sound source localization module to propose mask proposals, but how to ensure that it has universal capabilities?

 I noticed that the training data of OV-AVSS is AVS, which is divided into 40 classes as seen and 30 classes as unseen. However, previous algorithms, such as AVSegFormer, are also trained on AVS, which do not claim to be universal. I also noticed that the qualitative results given in the paper can indeed segment some unseen classes.

 Looking forward to receiving your reply, thank you in advance!

**Comment by ruohaoguo** [comment]
Thank you for raising this question.

During training, our universal sound source localization module not only associates audio with the appearance of visual objects for seen classes, but also aligns sounds with the spatial positions of these objects. This spatial alignment is not restricted by categories and is used to estimate which region of the image the sounding object is located in (e.g., the left area of the image). Therefore, such knowledge can be transferred to unseen classes to locate potential sounding objects. In addition,  temporal features (motion cues and changes in sound) can also assist in sound source localization. In fact, our goal is to imitate human ability: when people watch a musical instrument being played, even if they’ve never seen or heard it before, they can still tell where the sound is coming from.

However, as a pioneering work, the current approach is not perfect. Our model may easily overfit the base ones when both classes share similar appearances or sounds, since these classes are trained with higher confidence scores. Therefore, a more fine-grained feature discriminative modeling, including parts or attributes, is required to handle these problems. When audio-visual inconsistency and noisy background disturbances are involved, our model’s performance may be limited, leading to inaccurate segmentation results.

I hope this helps! Let me know if you have any further questions.

**Comment by ruohaoguo** [comment]
Following up on my previous comment, I’d like to add a few more thoughts.

Due to the utilization of the sound head instead of a class head, object queries are class-agnostic, enabling our model to detect objects from all categories. This universal object proposal has been widely proved in previous open-vocabulary works [1, 2]. For the mask prediction, the class-agnostic object queries are passed into a mask head and then dot-producted with the per-pixel embeddings, yielding the final masks.

[1] Haochen Wang, Cilin Yan, Shuai Wang, Xiaolong Jiang, Xu Tang, Yao Hu, Weidi Xie, and Efstratios Gavves. 2023. Towards open-vocabulary video instance segmentation. In Proceedings of the IEEE International Conference on Computer Vision. 4057–4066.
[2] Yuhang Zang, Wei Li, Kaiyang Zhou, Chen Huang, and Chen Change Loy. 2022. Open-vocabulary detr with conditional matching. In Proceedings of the Conference on European Conference on Computer Vision. 106–122.

**Comment by lchen1019** [comment]
This is very helpful. Thank you for your detailed response!
